integrations/Mouseflow/browser.js

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable no-underscore-dangle */
+/* eslint-disable class-methods-use-this */
+import { NAME } from "./constants";
+import logger from "../../utils/logUtil";
+import ScriptLoader from "../ScriptLoader";
+import { setCustomVariables, addTags } from "./utils";
+
+class Mouseflow {
+  constructor(config) {
+    this.websiteId = config.websiteId;
+    this.name = NAME;
+  }
+
+  init() {
+    logger.debug("===In init mouseflow===");
+    window._mfq = window._mfq || [];
+    ScriptLoader(
+      "mouseflow-integraions",
+      `https://cdn.mouseflow.com/projects/${this.websiteId}.js`
+    );
+  }
+
+  isLoaded() {
+    logger.debug("===In isLoaded mouseflow===");
+    return !!window.mouseflow && typeof window.mouseflow === "object";
+  }
+
+  isReady() {
+    logger.debug("===In isReady mouseflow===");
+    return !!window._mfq;
+  }
+
+  /**
+   * Identify.
+   * for supporting userId or email
+   * Ref: https://js-api-docs.mouseflow.com/#identifying-a-user
+   * for supporting user traits and tags
+   * Ref: https://js-api-docs.mouseflow.com/#setting-a-custom-variable
+   * @param {Identify} identify
+   */
+  identify(rudderElement) {
+    const { message } = rudderElement;
+    const { traits } = message.context;
+    const email = message.context.traits?.email || message.traits?.email;
+    const userId = message.userId || email;
+    window._mfq.push(["stop"]);
+    if (userId) window._mfq.push(["identify", userId]);
+    window.mouseflow.start();
+    setCustomVariables(traits);
+    addTags(message);
+  }
+
+  /**
+   * Track - tracks an event for a specific user
+   * for supporting event
+   * Ref: https://js-api-docs.mouseflow.com/#tagging-a-recording
+   * for supporting properties and tags
+   * Ref: https://js-api-docs.mouseflow.com/#setting-a-custom-variable
+   * @param {Track} track
+   */
+  track(rudderElement) {
+    logger.debug("=== In mouseflow track ===");
+
+    const { message } = rudderElement;
+    const { event, properties } = message;
+    if (!event) {
+      logger.error("[mouseflow]: Event name from track call is missing!!===");
+      return;
+    }
+    window._mfq.push(["tag", event]);
+    setCustomVariables(properties);
+    addTags(message);
```

Comment:
Why would anyone want to send one tag as the event name and the rest of them in the integrations object? @ItsSudip @utsabc.

Reply:
By default, we are storing the event name as a tag. And apart from that if someone wants to send more tags they can do through the integrations object.

Reply:
That doesn't make much sense. Is there a use case for this?

Reply:
There is not any endpoint to send track calls. Segment is sending the properties as key-value pairs. We thought of adding the event name as a tag will help them filter users.

Reply:
You can use tags for filtering/analysis. People might want to associate activity with custom tags, we have designed to allow users flexibility add custom tags.

Reply:
Treating the event name and tags from the integrations object as tags don't sound like a good idea. Why would anyone want to send tags in two places?

Only either of the following options makes sense to me:
1. Multiple track calls with valid event names with/without any properties to register event names as tags.
2. A single track call without event name but with/without properties and one or more tags in the integrations object.

Reply:
2. Is not really something we should go towards, event= mouseflow api recommends to set tags based on user activity which correlates more towards our event name

For passing tags in integrations object is allowing users to add custom tag to the session when this activity was  captured as tag.

We can see how we are envisioning this on mouseflow dashboard over a call would be easier to come to some alignment

Reply:
updated the tags to customVariables